staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -175,10 +175,31 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 	options := metav1.ListOptions{ResourceVersion: "0"}
 	r.metrics.numberOfLists.Inc()
 	start := r.clock.Now()
-	list, err := r.listerWatcher.List(options)
+
+	var list runtime.Object
+	var err error
+	listCh := make(chan struct{}, 1)
+	panicCh := make(chan interface{}, 1)
+	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				panicCh <- r
+			}
+		}()
+		list, err = r.listerWatcher.List(options)
+		close(listCh)
+	}()
+	select {
+	case <-stopCh:
+		return nil
+	case r := <-panicCh:
+		panic(r)
```

Comment:
note: this way you lose the backtrace of the actual issue.

Reply:
it's not easy to change that though. In the apiserver handler code we append errors with the backtrace string. Not nice either. Maybe just leave it here as it is.